relay-server/src/services/processor/replay.rs

```diff
@@ -55,13 +54,13 @@ pub fn process(
     let combined_envelope_items =
         project_state.has_feature(Feature::SessionReplayCombinedEnvelopeItems);
 
-    state.managed_envelope.retain_items(|item| {
-        // If replays aren't enabled or an item was dropped - drop the remainder of the
-        // envelope.
-        if !replays_enabled {
-            return ItemAction::DropSilently;
-        }
+    // If the replay feature is not enabled drop the items silenty.
+    if !replays_enabled {
+        state.managed_envelope.drop_items();
```

Comment:
This would also work, and would keep the API surface small. But `drop_items` is easier to read so I don't feel strongly about it:

```suggestion
        state
            .managed_envelope
            .retain_items(|_| ItemAction::DropSilently);
```

Reply:
I'd either rename to `drop_items_silently` or use `retain_items` to make it clear there are no side-effects.